amazon-personalize-ranking/src/main/java/org/opensearch/search/relevance/transformer/personalizeintelligentranking/reranker/impl/AmazonPersonalizedRankerImpl.java

```diff
@@ -96,7 +99,15 @@ public SearchHits rerank(SearchHits hits, PersonalizeRequestParameters requestPa
 
             SearchHits personalizedHits = combineScores(hits, result);
             return personalizedHits;
-        } catch (Exception ex) {
+        } catch (AmazonServiceException e) {
+            logger.error("Exception while calling personalize campaign: {}", e.getMessage());
+            int statusCode = e.getStatusCode();
+            if (statusCode >= 400 && statusCode < 500) {
```

Comment:
nit: Not a blocker but can we wrap this logic for identifying 4xx errors into an util method so that it can be reused in future?

Reply:
Updated in the revision